internal/etos/api/logarea.go

```diff
@@ -0,0 +1,241 @@
+// Copyright Axis Communications AB.
+//
+// For a full list of individual contributors, please see the commit history.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"context"
+	"fmt"
+
+	etosv1alpha1 "github.com/eiffel-community/etos/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	logAreaPort        int32 = 8080
+	LogAreaServicePort int32 = 80
+)
+
+type ETOSLogAreaDeployment struct {
+	etosv1alpha1.ETOSLogArea
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+// NewETOSLogAreaDeployment will create a new ETOS logarea reconciler.
+func NewETOSLogAreaDeployment(spec etosv1alpha1.ETOSLogArea, scheme *runtime.Scheme, client client.Client) (*ETOSLogAreaDeployment, error) {
+	return &ETOSLogAreaDeployment{spec, client, scheme}, nil
+}
+
+// Reconcile will reconcile the ETOS logarea to its expected state.
+func (r *ETOSLogAreaDeployment) Reconcile(ctx context.Context, cluster *etosv1alpha1.Cluster) error {
```

Comment:
General question, applies to all components: should there be more verbose info logging? Observability seems important here, could help a lot with debugging.

Reply:
We should need more yes. We should, however, identify the need first and add only relevant logs. The controller will easily get spammy.